packages/reactive-element/src/decorators/query.ts

```diff
@@ -98,15 +98,14 @@ export function query(selector: string, cache?: boolean): QueryDecorator {
             })();
       return desc(protoOrTarget, nameOrContext, {
         get(this: ReactiveElement): V {
-          if (cache) {
-            let result: V = get!.call(this);
-            if (result === undefined) {
-              result = doQuery(this);
+          let result: V = get!.call(this);
+          if (result === undefined) {
+            result = doQuery(this);
+            if (result || this.hasUpdated) {
```

Comment:
We try to always do an explicit comparison to avoid ToBoolean conversions:

```suggestion
            if (result !== null || this.hasUpdated) {
```